src/python/pants/backend/terraform/tool.py

```diff
@@ -30,5 +42,37 @@ def default_known_versions(cls):
         ]
 
 
+@dataclass(frozen=True)
+class TerraformProcess:
+    """A request to invoke Terraform."""
+
+    args: tuple[str, ...]
+    description: str
+    input_digest: Digest = EMPTY_DIGEST
+    output_files: tuple[str, ...] = ()
+
+
+@rule(level=LogLevel.DEBUG)
+async def setup_terraform_process(request: TerraformProcess, terraform: TerraformTool) -> Process:
+    downloaded_terraform = await Get(
+        DownloadedExternalTool,
+        ExternalToolRequest,
+        terraform.get_request(Platform.current),
+    )
+
+    input_digest = await Get(
+        Digest,
+        MergeDigests((request.input_digest, downloaded_terraform.digest)),
+    )
+
+    return Process(
+        argv=("./terraform",) + request.args,
+        input_digest=input_digest,
+        output_files=request.output_files,
+        description=request.description,
+        level=LogLevel.DEBUG,
```

Comment:
I think this is a sensible default, at least for now. Although might be worth eagerly adding `level` to `TerraformProcess` so callers are encouraged to set to INFO where appropriate. 

> I think this is a sensible default, at least for now.

Becauase we want `lint` and `fmt` Processes to be at Debug level, given that the core goals will already log the result for you at INFO or WARN level.

Reply:
Going to leave as-is then. If it needs INFO in the future, then can add a level field in.